File: backend/src/main/java/com/example/Flicktionary/global/security/CorsConfig.java
package com.example.Flicktionary.global.security;

import jakarta.servlet.http.HttpServletRequest;
import lombok.NonNull;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.CorsConfigurationSource;

import java.util.List;

@Configuration
public class CorsConfig implements CorsConfigurationSource {
    private static final String ALLOWED_ORIGIN = "http://localhost:3000";
    private static final List<String> ALLOWED_METHODS = List.of("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS");

    @Override
    public CorsConfiguration getCorsConfiguration(@NonNull HttpServletRequest request) {
        CorsConfiguration config = new CorsConfiguration();
        config.setAllowedOrigins(List.of(ALLOWED_ORIGIN));
        config.setAllowedMethods(ALLOWED_METHODS);
        config.setAllowCredentials(true);
        config.setAllowedHeaders(List.of("Authorization", "Content-Type", "Accept", "Origin"));
        config.setExposedHeaders(List.of("Authorization"));
        config.setMaxAge(3600L);

        return config;
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/global/security/CorsConfig.kt
package com.example.Flicktionary.global.security

import jakarta.servlet.http.HttpServletRequest
import lombok.NonNull
import org.springframework.context.annotation.Configuration
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.CorsConfigurationSource

@Configuration
class CorsConfig(): CorsConfigurationSource {
    private val ALLOWED_ORIGIN = "http://localhost:3000"
    private val ALLOWED_METHODS = listOf("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS")
    
    override fun getCorsConfiguration(@NonNull request: HttpServletRequest): CorsConfiguration {
        val config = CorsConfiguration()
        config.allowedOrigins = listOf(ALLOWED_ORIGIN)
        config.allowedMethods = ALLOWED_METHODS
        config.allowCredentials = true
        config.allowedHeaders = listOf("Authorization", "Content-Type", "Accept", "Origin")
        config.exposedHeaders = listOf("Authorization")
        config.maxAge = 3600L

        return config
    }
}